### PR TITLE
microblaze: apply gdb patches from meta-xilinx repository

### PR DESCRIPTION
**Pull request by alpsayin**
microblaze: apply gdb patches from meta-xilinx repository

This patchset fixes many known issues on gnu-toolchain for microblaze.
But mainly the atomic-cas and binutils incorrect relocation issues.

Patches obtained from https://github.com/xilinx/meta-xilinx/ repository,
from meta-microblaze/recipes-devtools/gdb path.

Introduction of Microblaze64 didn't break gdb build, so Ive decided to let it be.

### CODE REVIEW

File: Makefile.def [hunk 1]
```diff
@@ -120,7 +120,8 @@ host_modules= { module= libiconv;
 		missing= install-html;
 		missing= install-info; };
 host_modules= { module= m4; };
-host_modules= { module= readline; };
+host_modules= { module= readline;
+                extra_configure_flags='--enable-static';};
```
Comment (stephanosio) [inline Makefile.def]: This hunk does not seem to be relevant to adding microblaze support.

File: Makefile.in [hunk 2]
```diff
@@ -32791,7 +32791,7 @@ configure-readline:
 	  $$s/$$module_srcdir/configure \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(HOST_CONFIGARGS) --build=${build_alias} --host=${host_alias} \
-	  --target=${target_alias}  \
+	  --target=${target_alias} --enable-static \
```
Comment (stephanosio) [inline Makefile.in]: same here

File: sim/common/Make-common.in [hunk 3]
```diff
@@ -70,7 +70,7 @@ tooldir = $(libdir)/$(target_alias)
 datadir = @datadir@
 datarootdir = @datarootdir@
 mandir = @mandir@
-man1dir = $(mandir)/man1
+man1dir = $(DESTDIR)$(mandir)/man1
```
Comment (stephanosio) [inline sim/common/Make-common.in]: likely irrelevant

**Comment by alpsayin** [comment]
@stephanosio resolved the marked issues and some more that I noticed irrelevant to microblaze. Something bad happens during startup though. I'm inspecting this one further.
UPDATE:  it's the damn `microblaze64` or `linux-microblaze` ports again. I've filtered them out and kept one important patch that resolves a really annoying issue.

**Comment by alpsayin** [comment]
@stephanosio It's the damn microblaze64 or linux-microblaze ports again. I've filtered them out and kept one important patch that resolves a really annoying issue. Ready for re-review. Considering that only a single patch is left I'm half-expecting that I'll be asked to apply the [original commit](https://github.com/Xilinx/meta-xilinx/blob/master/meta-microblaze/recipes-devtools/gdb/gdb/0008-Patch-MicroBlaze.patch) with its author,date and message but with a bit of Zephyr commit guideline application `microblaze: foo bar etc.` ... ?

**Comment by stephanosio** [comment]
> @stephanosio It's the damn microblaze64 or linux-microblaze ports again. I've filtered them out and kept one important patch that resolves a really annoying issue. Ready for re-review. Considering that only a single patch is left I'm half-expecting that I'll be asked to apply the [original commit](https://github.com/Xilinx/meta-xilinx/blob/master/meta-microblaze/recipes-devtools/gdb/gdb/0008-Patch-MicroBlaze.patch) with its author,date and message but with a bit of Zephyr commit guideline application `microblaze: foo bar etc.` ... ?

Given that the original patch is fairly void of any useful information, the current commit looks good.

**Review by stephanosio — APPROVED** [review]
